src/main/java/sparksoniq/jsoniq/item/FunctionItem.java

```diff
@@ -145,7 +143,8 @@ public void read(Kryo kryo, Input input) {
         this.identifier = kryo.readObject(input, FunctionIdentifier.class);
         this.parameterNames = kryo.readObject(input, ArrayList.class);
         this.signature = kryo.readObject(input, ArrayList.class);
-        this.bodyExpression = kryo.readObject(input, Expression.class);
+        Class<? extends RuntimeIterator> iteratorClass = kryo.readObject(input, Class.class);
+        this.bodyIterator = kryo.readObject(input, iteratorClass);
```

Comment:
and here the reverse: read an array of byte[], and deserialize it with java.io.Serializable.

Reply:
or, if you really want to use Kryo, readClassAndObject